tests: lxml v6.0.0 breaks `test_xml_validates`

The release of [lxml](https://pypi.org/project/lxml) v6.0.0 on 26th June 2025 includes the change:

- Binary wheels use the library versions libxml2 2.14.4 and libxslt 1.1.43. Note that this disables direct HTTP and FTP support for parsing from URLs. Use Python URL request tools instead (which usually also support HTTPS).

The test `test_xml_validates` fails with an error message:
```
FAILED tests/doi_minter_test.py::test_xml_validates - lxml.etree.XMLSchemaParseError: Failed to locate the main schema resource at 'http://schema.datacite.org/meta/kernel-4.4/metadata.xsd'.
```
from the line:
https://github.com/HEPData/hepdata/blob/ab06dbc8ca230db7bc7e398494735cc3534ebc30/tests/doi_minter_test.py#L357

Using `requests.get` to download the schema is complicated by the fact that the schema contains lines like:

```xml
<xs:include schemaLocation="include/datacite-titleType-v4.xsd"/>
```

Rather than try to find a workaround using the [lxml](https://pypi.org/project/lxml) package, an alternative is to move to the [xmlschema](https://pypi.org/project/xmlschema) package instead.

--- a/hepdata/modules/records/api.py
+++ b/hepdata/modules/records/api.py
@@ -831,7 +831,8 @@ def check_and_convert_from_oldhepdata(input_directory, id, timestamp):
         successful = convert_oldhepdata_to_yaml(oldhepdata_found[1], converted_temp_path)
         if not successful:
             # Parse error message from title of HTML file, removing part of string after final "//".
-            soup = BeautifulSoup(open(converted_temp_path), "html.parser")
+            with open(converted_temp_path) as converted_temp_file:
+                soup = BeautifulSoup(converted_temp_file, "html.parser")
             errormsg = soup.title.string.rsplit("//", 1)[0]
 
     except Error as error:  # hepdata_converter_ws_client.Error

--- a/hepdata/modules/records/api.py
+++ b/hepdata/modules/records/api.py
@@ -831,7 +831,7 @@ def check_and_convert_from_oldhepdata(input_directory, id, timestamp):
         successful = convert_oldhepdata_to_yaml(oldhepdata_found[1], converted_temp_path)
         if not successful:
             # Parse error message from title of HTML file, removing part of string after final "//".
-            soup = BeautifulSoup(open(converted_temp_path), "lxml")
+            soup = BeautifulSoup(open(converted_temp_path), "html.parser")
             errormsg = soup.title.string.rsplit("//", 1)[0]
 
     except Error as error:  # hepdata_converter_ws_client.Error

--- a/hepdata/version.py
+++ b/hepdata/version.py
@@ -28,4 +28,4 @@ This file is imported by ``HEPData.__init__``,
 and parsed by ``setup.py``.
 """
 
-__version__ = "0.9.4dev20250506"
+__version__ = "0.9.4dev20250702"

--- a/setup.py
+++ b/setup.py
@@ -39,7 +39,7 @@ tests_require = [
     'pytest-timeout>=1.4.2',
     'requests-mock>=1.8.0',
     'selenium>=4.0.0',
-    'xmlschema'
+    'xmlschema>=4.0.0'
 ]
 
 extras_require = {

--- a/setup.py
+++ b/setup.py
@@ -39,7 +39,7 @@ tests_require = [
     'pytest-timeout>=1.4.2',
     'requests-mock>=1.8.0',
     'selenium>=4.0.0',
-    'lxml'
+    'xmlschema'
 ]
 
 extras_require = {

--- a/tests/doi_minter_test.py
+++ b/tests/doi_minter_test.py
@@ -7,7 +7,7 @@ from flask import render_template
 from invenio_db import db
 from invenio_pidstore.models import PersistentIdentifier
 from invenio_pidstore.errors import PIDInvalidAction, PIDDoesNotExistError
-import lxml
+import xmlschema
 import pytest
 
 from hepdata.modules.records.importer.api import import_records
@@ -354,7 +354,7 @@ def test_xml_validates(app, identifiers):
     site_url = app.config.get('SITE_URL', 'https://www.hepdata.net')
 
     # Load schema
-    datacite_schema = lxml.etree.XMLSchema(file = 'http://schema.datacite.org/meta/kernel-4.4/metadata.xsd')
+    datacite_schema = xmlschema.XMLSchema('http://schema.datacite.org/meta/kernel-4.4/metadata.xsd')
 
     base_xml = render_template('hepdata_records/formats/datacite/datacite_container_submission.xml',
                                doi=hep_submission.doi,
@@ -362,8 +362,8 @@ def test_xml_validates(app, identifiers):
                                data_submissions=data_submissions,
                                publication_info=publication_info,
                                site_url=site_url)
-    base_doc = lxml.etree.fromstring(base_xml)
-    datacite_schema.assertValid(base_doc)
+    # Validate the base XML
+    datacite_schema.validate(base_xml)
 
     version_xml = render_template('hepdata_records/formats/datacite/datacite_container_submission.xml',
                                   doi=f"{hep_submission.doi}.v1",
@@ -371,8 +371,8 @@ def test_xml_validates(app, identifiers):
                                   data_submissions=data_submissions,
                                   publication_info=publication_info,
                                   site_url=site_url)
-    version_doc = lxml.etree.fromstring(version_xml)
-    datacite_schema.assertValid(version_doc)
+    # Validate the version XML
+    datacite_schema.validate(version_xml)
 
     for data_submission in data_submissions:
         data_xml = render_template('hepdata_records/formats/datacite/datacite_data_record.xml',
@@ -383,8 +383,8 @@ def test_xml_validates(app, identifiers):
                                    data_submission=data_submission,
                                    publication_info=publication_info,
                                    site_url=site_url)
-        data_doc = lxml.etree.fromstring(data_xml)
-        datacite_schema.assertValid(data_doc)
+        # Validate the data submission XML
+        datacite_schema.validate(data_xml)
 
     # Import a record with a resource file
     import_records(['ins1748602'], synchronous=True)
@@ -408,10 +408,8 @@ def test_xml_validates(app, identifiers):
             publication_info=publication_info,
             site_url=site_url)
 
-        print(resource_xml)
-
-        doc = lxml.etree.fromstring(resource_xml)
-        datacite_schema.assertValid(doc)
+        # Validate the resource XML
+        datacite_schema.validate(resource_xml)
 
 
 def test_get_submission_file_resources(app, identifiers):

